### PR TITLE
chore(cz): update commitizen (cz) and add scripts around it

### PR DESCRIPTION
**Pull request by ctrlc03**
This is in reference to us establishing our GH processes (#847). Users can easily commit using convential messages in an interactive way by running:

`npm run commit`

Furthermore, pre-commit hooks will ensure that the commit message follows the guideline.

### CODE REVIEW

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    extends: ['@commitlint/config-conventional'],
+  };
```
Comment (0xmad) [inline commitlint.config.js:3]: Please add new line at the end

File: package.json [hunk 2]
```diff
@@ -42,5 +45,12 @@
     "commitizen": {
       "path": "./node_modules/cz-conventional-changelog"
     }
+  },
+  "husky": {
+    "hooks": {
+      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS",
+      "pre-commit": "lint-staged"
+    }
   }
```
Comment (0xmad) [inline package.json:54]: I don't think this is needed here. We have `.husky` folder with all the githooks

**Review by 0xmad — CHANGES_REQUESTED** [review]
@ctrlc03 thanks, just few comments.

File: .eslintignore [hunk 3]
```diff
@@ -4,3 +4,4 @@ coverage
 build
 typechain-types
 .eslintrc.js
+commitlint.config.js
```
Comment (0xmad) [inline .eslintignore:7]: And new line at the end here

Comment (ctrlc03) [reply .eslintignore:7]: 😤

**Review by 0xmad — APPROVED** [review]
@ctrlc03 thanks!

**Pull request merged**